examples/gcoap/client.c

```diff
@@ -146,7 +146,8 @@ static void _resp_handler(const gcoap_request_memo_t *memo, coap_pkt_t* pdu,
 
             int len = coap_opt_finish(pdu, COAP_OPT_FINISH_NONE);
             gcoap_req_send((uint8_t *)pdu->hdr, len, remote,
-                           _resp_handler, memo->context);
+                           _resp_handler, memo->context,
+                           GCOAP_SOCKET_TYPE_UNDEF);
```

Comment:
I do not understand this change in the context of your PR description.

Reply:
This PR makes `gcoap_req_send()` a full alias of `gcoap_req_send_tl()` (or rather vice-versa). Up until now, `gcoap_req_send()` was a wrapper around `gcoap_req_send_tl()` with `tl_type` being set to a default parameter (`GCOAP_SOCKET_TYPE_UNDEF`). With `gcoap_req_send()` now becoming a full alias for `gcoap_req_send_tl()` all calls to `gcoap_req_send()` need to have this parameter provided.

Reply:
Updated the faulty PR description ;-).